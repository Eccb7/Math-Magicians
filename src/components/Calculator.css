.calculator-container {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
}

.calculator {
  width: 300px;
  border: 1px solid #ccc;
  border-radius: 5px;
  padding: 30px;
  background-color: #b8b6b6;
}

.display {
  height: 50px;
  line-height: 50px;
  text-align: right;
  font-size: 24px;
  background-color: #464545;
  border: 1px solid #ccc;
  border-radius: 5px;
  margin-bottom: 10px;
  padding: 0 10px;
}

.col {
  padding: 0;
}

.button-gray {
  background-color: rgb(182, 179, 179);
  color: white;

  /* padding: 8px; */
}

.button-orange {
  background-color: rgb(250, 187, 69);
  color: white;

  /* padding: 8px; */
}

.result-input {
  background-color: #666363;
  color: #fff;
  padding: 10px;
  margin: 0;

  /* box-shadow: #666363; */
}
