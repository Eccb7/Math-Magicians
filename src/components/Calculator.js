import React, { useState } from 'react';
import './Calculator.css';

const Calculator = () => {
  const [result, setResult] = useState('0');

  const handleClick = (value) => {
    setResult((prevResult) => {
      if (prevResult === '0') {
        return value;
      }
      return prevResult + value;
    });
  };

  const handleCalculate = () => {
    try {
    //   eslint-disable-next-line
      setResult(eval(result).toString());
    } catch (error) {
      setResult('Error');
    }
  };

  const handleClear = () => {
    setResult('0');
  };

  return (
    <div className="calculator-container">
      <div className="calculator">
        <input type="text" value={result} readOnly className="result-input w-100" />
        <div className="keypad">
          <div className="row">
            <div className="col">
              <button type="button" onClick={handleClear} className="button-gray w-100">AC</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('+/-')} className="button-orange w-100">+/-</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('%')} className="button-orange w-100">%</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('/')} className="button-orange w-100">/</button>
            </div>
          </div>
          <div className="row">
            <div className="col">
              <button type="button" onClick={() => handleClick('7')} className="button-gray w-100">7</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('8')} className="button-gray w-100">8</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('9')} className="button-gray w-100">9</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('-')} className="button-orange w-100">-</button>
            </div>
          </div>
          <div className="row">
            <div className="col">
              <button type="button" onClick={() => handleClick('4')} className="button-gray w-100">4</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('5')} className="button-gray w-100">5</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('6')} className="button-gray w-100">6</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('*')} className="button-orange w-100">x</button>
            </div>
          </div>
          <div className="row">
            <div className="col">
              <button type="button" onClick={() => handleClick('1')} className="button-gray w-100">1</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('2')} className="button-gray w-100">2</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('3')} className="button-gray w-100">3</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('+')} className="button-orange w-100">+</button>
            </div>
          </div>
          <div className="row">
            <div className="col">
              <button type="button" onClick={() => handleClick('0')} className="button-gray w-100">0</button>
            </div>
            <div className="col">
              <button type="button" onClick={() => handleClick('.')} className="button-gray w-100">.</button>
            </div>
            <div className="col">
              <button type="button" onClick={handleCalculate} className="button-orange w-100">=</button>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Calculator;
