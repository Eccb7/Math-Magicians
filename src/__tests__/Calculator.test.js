import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('Calculator', () => {
  test('should update calcData when buttons are clicked', () => {
    // Render the Calculator component
    const { getByText, getByTestId } = render(<Calculator />);

    // Click on buttons and verify state updates
    fireEvent.click(getByText('1'));
    expect(getByTestId('result-input')).toHaveValue('1');

    fireEvent.click(getByText('+'));
    expect(getByTestId('result-input')).toHaveValue('1');

    fireEvent.click(getByText('2'));
    expect(getByTestId('result-input')).toHaveValue('2');

    fireEvent.click(getByText('='));
    expect(getByTestId('result-input')).toHaveValue('3');

    fireEvent.click(getByText('x'));
    expect(getByTestId('result-input')).toHaveValue('3');

    fireEvent.click(getByText('4'));
    expect(getByTestId('result-input')).toHaveValue('4');

    fireEvent.click(getByText('='));
    expect(getByTestId('result-input')).toHaveValue('12');

    fireEvent.click(getByText('AC'));
    expect(getByTestId('result-input')).toHaveValue('');
  });

  test('should match the snapshot', () => {
    // Render the Calculator component
    const { asFragment } = render(<Calculator />);

    // Create a snapshot and compare it to the stored snapshot
    expect(asFragment()).toMatchSnapshot();
  });
});
