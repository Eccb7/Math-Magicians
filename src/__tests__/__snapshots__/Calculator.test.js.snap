// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator-container"
  >
    <div
      class="math"
    >
      LETS DO SOME MATH
    </div>
    <div
      class="calculator"
    >
      <input
        class="result-input w-100"
        data-testid="result-input"
        readonly=""
        type="text"
        value=""
      />
      <div
        class="keypad"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              AC
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-orange w-100"
              type="button"
            >
              +/-
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-orange w-100"
              type="button"
            >
              %
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-orange w-100"
              type="button"
            >
              /
            </button>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              7
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              8
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              9
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-orange w-100"
              type="button"
            >
              -
            </button>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              4
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              5
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              6
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-orange w-100"
              type="button"
            >
              x
            </button>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              1
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              2
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              3
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-orange w-100"
              type="button"
            >
              +
            </button>
          </div>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              0
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-gray w-100"
              type="button"
            >
              .
            </button>
          </div>
          <div
            class="col"
          >
            <button
              class="button-orange w-100"
              type="button"
            >
              =
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
